tests/suggestion-box.test.ts

```diff
@@ -0,0 +1,309 @@
+                </aalam-sgn-box>
+            `
+        );
+        const input = el.shadowRoot?.querySelector("input");
+        el.list = [
+            {
+                name: "thiruvarur",
+                description: "My native",
+            },
+            {
+                name: "coimbatore",
+                description: "Working place",
+            },
+            {
+                name: "chennai",
+                description: "dream location",
+            },
+        ];
+        el.listkey = "name";
+        input.value = "thi";
+        input.dispatchEvent(
+            new Event("input", { bubbles: true, composed: true })
+        );
+        await el.updateComplete;
+
+        expect(el.filtered_list.length).to.equal(1);
+        expect(el.filtered_list[0].name).to.equal("thiruvarur");
+    });
+
+    it("should update suggestions using setSuggestion", async () => {
+        const el = await fixture(
+            html`
+                <aalam-sgn-box>
+                    <div slot="sgn-item-template">{name}</div>
+                </aalam-sgn-box>
+            `
+        );
+        const input = el.shadowRoot?.querySelector("input");
+        input.value = "thi";
+        input.dispatchEvent(
+            new Event("input", { bubbles: true, composed: true })
+        );
+
+        // Simulate a server response or any other source of suggestions
+        const suggestions = [
+            {
+                name: "thiruvarur",
+                description: "My native",
+            },
+        ];
+
+        el.setSuggestion(suggestions, true);
+        await el.updateComplete;
+
+        expect(el.filtered_list.length).to.equal(1);
+        expect(el.filtered_list[0].name).to.equal("thiruvarur");
+    });
+
+    it("should navigate suggestions with keyboard", async () => {
+        // keyboard event arrow down and arrow up
+        const el = await fixture(
+            html`
+                <aalam-sgn-box
+                    .list=${[{ name: "test1" }, { name: "test2" }]}
+                    .listkey=${"name"}
+                >
+                    <div slot="sgn-item-template">{name}</div>
+                </aalam-sgn-box>
+            `
+        );
+
+        const input = el.shadowRoot?.querySelector("input");
+        if (!input) throw new Error("Input element not found");
+
+        input.value = "test";
+        input.dispatchEvent(
+            new Event("input", { bubbles: true, composed: true })
+        );
+        await el.updateComplete;
+
+        input.focus();
+        await triggerFocusFor(input);
+
+        const arrowDownEvent = new KeyboardEvent("keydown", {
+            key: "ArrowDown",
+            bubbles: true,
+            composed: true,
+        });
+        input.dispatchEvent(arrowDownEvent);
+        await el.updateComplete;
+
+        expect(el.index).to.equal(0);
+
+        input.dispatchEvent(arrowDownEvent);
+        await el.updateComplete;
+
+        expect(el.index).to.equal(1);
```

Comment:
Use sendKeys command. Refer datepicker test case.

You can test much more in this. Set suggestion with 7/8 results and continuously navigate the arrow keys up and down and check if they are looping. You can check with and without load more.